Disable FORTIFY_SOURCE check for length of string in unget_wch.
Leave the rest of the code as is.

---
 ncurses/widechar/lib_unget_wch.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ncurses/widechar/lib_unget_wch.c b/ncurses/widechar/lib_unget_wch.c
--- a/ncurses/widechar/lib_unget_wch.c
+++ b/ncurses/widechar/lib_unget_wch.c
@@ -79,7 +79,7 @@ NCURSES_SP_NAME(unget_wch) (NCURSES_SP_D
 
     if (length != (size_t) (-1)
 	&& length != 0) {
-	char *string;
+	char * volatile string;
 
 	if ((string = (char *) malloc(length)) != NULL) {
 	    int n;
